### PR TITLE
Feature. Orchestrator & backend: Retrieve batches of test IDs from DB

### LINKED ISSUE #51: Orchestrator & backend: Retrieve batches of test IDs from DB
Just batches of constant size, dynamic scaling will be added later.

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Sending batches logic added.
  * Added tests

This PR closes #51

### CODE REVIEW

File: db/changelog/v-1/11-changeset-test-agent-insert-data.xml [hunk 1]
```diff
@@ -25,4 +37,24 @@
         </insert>
     </changeSet>
 
+    <changeSet id="test-insert" author="aktsay" context="dev">
```
Comment (petertrr) [inline db/changelog/v-1/11-changeset-test-agent-insert-data.xml:40]: Why so many changesets?

File: save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TestRepository.kt [hunk 2]
```diff
@@ -2,8 +2,17 @@ package org.cqfn.save.backend.repository
...
  * Repository of tests
  */
-interface TestRepository : JpaRepository<Test, String>
+interface TestRepository : JpaRepository<Test, String> {
+    /**
+     * Method to retrieve ready batches
+     *
+     * @return List of Tests
+     */
+    @Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY'", nativeQuery = true)
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TestRepository.kt:16]: Batch should also be limited by a fixed size (let's say 20 ids for now), and TestService should store offset for the last test

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt [hunk 3]
```diff
@@ -15,6 +15,8 @@ class TestService {
     @Autowired
     private lateinit var testRepository: TestRepository
 
+    private var offset = 0
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt:18]: AtomicInteger?

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt [hunk 4]
```diff
@@ -25,7 +27,11 @@ class TestService {
...
      * @return Test batches
      */
-    // Fixme: Stub in here
-    fun getTestBatches() = Mono.just(listOf(TestDto("qwe", "www", 0, "id"),
-        TestDto("qwe", "www", 0, "qweqwe")))
+    fun getTestBatches(): Mono<List<TestDto>> {
+        val tests = testRepository.retrieveBatches(offset).map {
+            TestDto(it.expectedFilePath, it.testFilePath, it.testSuiteId, it.id)
+        }
+        offset += 20
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt:34]: This value is duplicated twice, you should extract it into a constant. I believe, `const val`s are allowed for annotation parameters.

File: save-backend/src/test/kotlin/org/cqfn/save/backend/controller/TestInitializeControllerTest.kt [hunk 5]
```diff
@@ -67,4 +69,17 @@ class TestInitializeControllerTest {
 
         assertTrue(databaseData.any { it.id == test.id && it.expectedFilePath == test.expectedFilePath })
     }
+
+    @Test
+    fun checkServiceData() {
```
Comment (petertrr) [inline save-backend/src/test/kotlin/org/cqfn/save/backend/controller/TestInitializeControllerTest.kt:74]: Can we insert 21 executions into DB here to test if batching works as expected?

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt (hunk 3 above)
Comment (aktsay6) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt]: Fixed

File: save-backend/src/test/kotlin/org/cqfn/save/backend/controller/TestInitializeControllerTest.kt (hunk 5 above)
Comment (aktsay6) [reply save-backend/src/test/kotlin/org/cqfn/save/backend/controller/TestInitializeControllerTest.kt:74]: Tested

File: save-backend/src/test/resources/application.properties [hunk 6]
```diff
@@ -1,5 +1,7 @@
 server.port = 5000
 backend.preprocessorUrl=http://preprocessor:5200
+backend.limit=20
+backend.offset=20
```
Comment (petertrr) [inline save-backend/src/test/resources/application.properties:4]: Why are these two separate parameters?

**Review by petertrr — APPROVED** [review]

**Pull request merged**